src/components/ContactUs/index.js

```diff
@@ -0,0 +1,194 @@
+                        value={values.phone}
+                      />
+                      {errors.phone && touched.phone && (
+                        <ErrorMsg>{errors.phone}</ErrorMsg>
+                      )}
+                    </td>
+                    <td>
+                      <label htmlFor="email">Email:</label>
+                      <ContactInput
+                        type="text"
+                        id="email"
+                        placeholder="jane-doe@example.com"
+                        name="email"
+                        value={values.email}
+                        onChange={handleChange}
+                        onBlur={handleBlur}
+                      />
+                      {errors.email && touched.email && (
+                        <ErrorMsg>{errors.email}</ErrorMsg>
+                      )}
+                    </td>
+                  </tr>
+                </tbody>
+              </ContactUsTable>
+              <label htmlFor="message">Message:</label>
+              <ContactTextArea
+                type="text"
+                id="message"
+                onChange={handleChange}
+                name="msg"
+                row="5"
+                value={values.msg}
+                onBlur={handleBlur}
+              />
+              {errors.msg && touched.msg && <ErrorMsg>{errors.msg}</ErrorMsg>}
+              <br />
+              <ContactUsBtn type="submit">Submit</ContactUsBtn>
+              <ContactCloseBtn onClick={props.click}>Exit</ContactCloseBtn>
+            </ContactForm>
+          </div>
+        )}
+      </ContactFormContainer>
+    </ContactBackground>
+  );
+};
+
+const EnhancedWithFormik = withFormik({
+  mapPropsToValues: () => ({
+    firstName: "",
+    lastName: "",
+    phone: "",
+    email: "",
+    msg: ""
+  }),
+
+  validate: values => {
+    const errors = {};
+
+    if (!values.email) {
+      errors.email = "Required";
+    } else if (
+      !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)
+    ) {
+      errors.email = "Invalid email address";
+    }
+
+    if (!values.firstName) {
+      errors.firstName = "Required";
+    }
+
+    if (!values.lastName) {
+      errors.lastName = "Required";
+    }
+
+    if (!values.phone) {
+      errors.phone = "Required";
+    } else if (!/^\D?(\d{3})\D?\D?(\d{3})\D?(\d{4})$/.test(values.phone)) {
+      errors.phone = "Invalid number";
+    }
+
+    if (!values.msg) {
+      errors.msg = "Required";
+    }
+
+    return errors;
+  },
+
+  handleSubmit: values => {
+    setTimeout(() => {
+      let data = "";
+      for (let key in values) {
+        data = `${data} \`${key}\`: ${values[key]}`;
+      }
+      const body = JSON.stringify({ text: data });
+      const url =
+        "https://hooks.slack.com/services/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";
+      fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/x-www-form-urlencoded"
```

Comment:
when you have to do weird things like what we are doing here we should add a comment that states why we are doing and a link to any reference ie the slack stackoverflow.